data/lib/service/match/match_service.dart

```diff
@@ -532,18 +532,27 @@ class MatchService {
     List<MatchTeamModel> teamList,
   ) async {
     try {
-      final List<MatchTeamModel> teams = [];
-
-      await Future.forEach(teamList, (e) async {
-        final TeamModel team = await _teamService.getTeamById(e.team_id);
-        final List<MatchPlayer> squad =
-            await getPlayerListFromPlayerIds(e.squad);
-        teams.add(
-          e.copyWith(team: team, squad: squad),
+      final teamIds = teamList.map((e) => e.team_id).toList();
+
+      final teamsData = await Future.wait([
+        _teamService.getTeamsByIds(teamIds),
+        Future.wait(
+          teamList
+              .map((team) => getPlayerListFromPlayerIds(team.squad))
+              .toList(),
+        ),
+      ]);
```

Comment:
can we optimize it to not fetch unwanted data?

getTeamByIds internally fetches players of team and get PlayerListFromPlayerIDs also fetches players which is in squad right?

just think like:  it's better to user contains and where kind of operation than fetching data from firestore as it'll consume more time.